### PR TITLE
Implement basic Open Graph support.

### PR DESCRIPTION
**Pull request by minecrafter**
This PR adds support for the [Open Graph protocol](http://ogp.me/).

### CODE REVIEW

File: models/user.go [hunk 1]
```diff
@@ -252,6 +252,11 @@ func (u *User) HomeLink() string {
 	return setting.AppSubURL + "/" + u.Name
 }
 
+// HTMLURL returns the user or organization's full link.
+func (u *User) HTMLURL() string {
+	return setting.AppURL + u.Name
```
Comment (lunny) [inline models/user.go:257]: Why not use `HomeLink()`?

**Review by lunny — CHANGES_REQUESTED** [review]

File: models/user.go (hunk 1 above)
Comment (minecrafter) [reply models/user.go:257]: `HomeLink()` returns a relative URL, and Open Graph demands a canonical URL.

Comment (lunny) [reply models/user.go:257]: OK

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
LGTM

**Review by appleboy — APPROVED** [review]

**Comment by appleboy** [comment]
LGTM Nice feature.

**Pull request merged**